src/Agent.Worker/TestResults/TestRunPublisher.cs

```diff
@@ -154,6 +160,34 @@ public TestRunData ReadResultsFromFile(TestRunContext runContext, string filePat
         }
         #endregion
 
+        private bool HierarchicalSubresultPreProcessing(string automatedTestName, List<TestCaseSubResultData> subResults, int level = 1)
+        {
+            int maxSubResultHierarchyLevel = TestManagementConstants.maxHierarchyLevelForSubresults;
+            int maxSubResultIterationCount = TestManagementConstants.maxSubResultPerLevel;
+            if (subResults == null || !subResults.Any())
```

Comment:
`subResults.Count == 0` instead of `!subResults.Any()`, LINQ is slower in this simple operation.